cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -64,6 +64,10 @@ private String parse(String str, int depth) throws InvalidMacroException {
 
     MacroMetadata macroPosition = findRightmostMacro(str);
     while (macroPosition != null) {
+      // the DefaultMacroEvaluator would throw an exception earlier, but iif a MacroEvaluator doesn't this is needed
```

Comment:
should not mentioned DefaultMacroEvaluator, there is no such class in the cdap platform. 

also, iif -> if